src/api/legendSets.js

```diff
@@ -74,11 +74,14 @@ export const apiFetchLegendSetsByDimension = async ({
         default:
             throw new Error(`${dimensionType} is not a valid dimension type`)
     }
+
+    const [id] = dimensionId.split('.').reverse()
```

Comment:
This was actually based on @edoardo 's way of reading the `programStageId` and the `dimensionId` from a string that can be either "dimensionId" or "programStageId.dimensionId", but since I don't need the `programStageId` I just omitted that part of the destruction. Maybe not the most readable, but it's short! 🙂